### DISCUSSION TITLE
Performing a join on a table before streaming it to a client in real time

### DISCUSSION THREAD

**Discussion by otterlistic** [supabase/supabase#2615] [Questions]
The title sums up what I'm trying to do, but here's a scenario:
table entities -> id, name
table entities_log -> id, uid, type, etc. etc.

what I need to do is, listen to all the new inserts of entities_log, but I want to perform a join so that I get the name of the entity alongside the inserted data. I attempted to solve this by creating a view, and then listening to changes on that view but nothing is being streamed to the client in real time. 

I'm open for any and all suggestions, because right now my only hope is doing that join in the client side which is just a big no no.

**Comment by dshukertjr** [comment]
Hi @Symasia !

At the moment, realtime on views is not supported on Supabase. I also would like to have this feature, but in the mean while, you can listen to changes from both `entities` and `entities_log`, and combine the data on the client side to simulate a join.

**Reply by otterlistic** [reply]
yeah ik, i think it's conceptually impossible too because views are not tables because i believe currently realtime makes use of publications and replication to pull everything together. kinda sucks, because this is a common issue with firestore too.

therefore i think the most ideal solution is probably enhancing the flexibility of the query builder for realtime
